crates/move-stackless-bytecode/src/conditional_merge_insertion.rs

```diff
@@ -76,6 +76,24 @@ impl ConditionalMergeInsertionProcessor {
             .map(|(label, _)| *label)
     }
 
+    fn is_side_effecting_op(op: &Operation) -> bool {
+        use Operation::*;
+        matches!(
+            op,
+            MoveTo(..)
+                | MoveFrom(..)
+                | BorrowGlobal(..)
+                | GetGlobal(..)
+                | WriteRef
+                | Havoc(..)
+                | Stop
+                | Uninit
+                | Destroy
+                | EmitEvent
+                | EventStoreDiverge
```

Comment:
Not sure, e.g., how much we care to reject 'EmitEvent' calls as being side-effecting, but at least now it's explicit what we accent/reject.